clusterctl/examples/openstack/provider-components.yaml.template

```diff
@@ -0,0 +1,292 @@
+    spec:
+      nodeSelector:
+        node-role.kubernetes.io/master: ""
+      tolerations:
+      - effect: NoSchedule
+        key: node-role.kubernetes.io/master
+      - key: CriticalAddonsOnly
+        operator: Exists
+      - effect: NoExecute
+        key: node.alpha.kubernetes.io/notReady
+        operator: Exists
+      - effect: NoExecute
+        key: node.alpha.kubernetes.io/unreachable
+        operator: Exists
+      containers:
+      - name: controller-manager
+        image: gcr.io/k8s-cluster-api/controller-manager:0.0.7
+        volumeMounts:
+          - name: config
+            mountPath: /etc/kubernetes
+          - name: certs
+            mountPath: /etc/ssl/certs
+        command:
+        - "./controller-manager"
+        args:
+        - --kubeconfig=/etc/kubernetes/admin.conf
+        resources:
+          requests:
+            cpu: 100m
+            memory: 20Mi
+          limits:
+            cpu: 100m
+            memory: 30Mi
+      - name: openstack-machine-controller
+        image: gcr.io/openstack-cluster-api/openstack-machine-controller:0.0.1
+        volumeMounts:
+          - name: config
+            mountPath: /etc/kubernetes
+          - name: certs
+            mountPath: /etc/ssl/certs
+          - name: sshkeys
+            mountPath: /etc/sshkeys
+          - name: machine-setup
+            mountPath: /etc/machinesetup
+          - name: cloudConfig
+            mountPath: /etc/cloud
+          - name: kubeadm
+            mountPath: /usr/bin/kubeadm
+        command:
+        - "./openstack-machine-controller"
+        args:
+        - --kubeconfig=/etc/kubernetes/admin.conf
+        resources:
+          requests:
+            cpu: 100m
+            memory: 20Mi
+          limits:
+            cpu: 100m
+            memory: 30Mi
+      volumes:
+      - name: config
+        hostPath:
+          path: /etc/kubernetes
+      - name: certs
+        hostPath:
+          path: /etc/ssl/certs
+      - name: sshkeys
+        secret:
+          secretName: machine-controller-sshkeys
+          defaultMode: 256
+      - name: cloudConfig
+        secret:
+          secretName: cloud-config
+      - name: machine-setup
+        configMap:
+          name: machine-setup
+      - name: kubeadm
+        hostPath:
+          path: /usr/bin/kubeadm
+---
+apiVersion: v1
+kind: Secret
+type: Opaque
+metadata:
+  name: machine-controller-sshkeys
+  namespace: default
+data:
+  user: $MACHINE_CONTROLLER_SSH_USER
+  private: $MACHINE_CONTROLLER_SSH_PRIVATE
+  public: $MACHINE_CONTROLLER_SSH_PUBLIC
+---
+apiVersion: v1
+kind: Secret
+type: Opaque
+metadata:
+  name: cloud-config
+  namespace: default
+data:
+  config_config.yaml: $OPENSTACK_CLOUD_CONFIG
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: machine-setup
+data:
+  machine_setup_configs.yaml: |-
+    items:
+    - machineParams:
+      - os: ubuntu-1604-lts
+        roles:
+        - Master
+        versions:
+          kubelet: 1.9.4
```

Comment:
can we switch to latest 1.10.x kubernetes? is that easy to do?

Reply:
Yes, we can do that. This file is a template for user can make their only configuration accordingly. Theoretically, they can change this to any version they need.